backend/api/Repositories/RestCommon/BaseRestRespository.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Pims.Api.Models;
+
+namespace Pims.Api.Repositories.Rest
+{
+    /// <summary>
+    /// BaseRestRepository provides common methods to interact with Rest-ful external interfaces.
+    /// </summary>
+    public abstract class BaseRestRepository : IRestRespository
+    {
+        protected readonly IHttpClientFactory _httpClientFactory;
+        protected readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BaseRestRepository"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="httpClientFactory">Injected Httpclient factory.</param>
+        protected BaseRestRepository(
+            ILogger logger,
+            IHttpClientFactory httpClientFactory)
+        {
+            _logger = logger;
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public abstract void AddAuthentication(HttpClient client, string authenticationToken = null);
+
+        public async Task<ExternalResult<T>> GetAsync<T>(Uri endpoint, string authenticationToken)
+        {
+            using HttpClient client = _httpClientFactory.CreateClient("Pims.Api.Logging");
```

Comment:
could we put this common code into a private function to cut down on code duplication?

Reply:
We cant. The 'using' statement needs to be in the scope of the function that will dispose it